### ISSUE TITLE
Program hangs after entering username / auth token / channel ID

### ISSUE THREAD

**Issue by ek1506** [elevenchars/discorddelete#2]
After entering the requested information and hitting enter, the program hangs with a blinking cursor and I don't get any messages, and no discord messages are deleted. I have tried using my username both with and without its numeric ID (#XXXX). Note that I do not have moderator privileges in this channel, I am just trying to delete my own messages.

Screenshot of the issue:

![shell](https://user-images.githubusercontent.com/29785364/28454236-480cc95e-6daf-11e7-9543-f935824a9f60.png)

**Comment by ek1506** [comment]
As a follow up, the hanging was just a result of waiting for the program to scan all the messages. After about ~10 minutes, it seemed to finish but I ended up with the following error message, and no deletions from discord:
"RuntimeError: maximum recursion depth exceeded while calling a Python object"

![uiojnouihjnuio](https://user-images.githubusercontent.com/29785364/28455194-02859c4a-6db3-11e7-8934-2de23dc0d75e.png)

**Comment by elevenchars** [comment]
Is this a large channel? After doing some googling it seems the only chance it could happen is if the channel has > 100000 messages. Tonight I'll start working on switching away from a recursive function because i was not aware of the limitations of recursive calls in python.

**Comment by ek1506** [comment]
Yes, it's a very large channel easily over 100k messages. I think you're right.

**Comment by amuse11** [comment]
Will this allow users who aren't mods to use this program?

**Comment by taofd** [comment]
Also getting error:

`RuntimeError: maximum recursion depth exceeded while calling a Python object`

In addition, the json doesn't seem to be working (gives me an auth_token not defined error) but not using the json and manually entering the parameters works fine. until it gets the above error :cry: .
